erpnext/stock/doctype/item/item.py

```diff
@@ -656,6 +659,27 @@ def after_rename(self, old_name, new_name, merge):
 					frappe.db.set_value(dt, d.name, "item_wise_tax_detail",
 											json.dumps(item_wise_tax_detail), update_modified=False)
 
+	def validate_duplicate_item_in_stock_reconciliation(self, old_name, new_name):
+		records = frappe.db.sql(""" SELECT parent, COUNT(*) as records
+			FROM `tabStock Reconciliation Item`
+			WHERE item_code = %s and docstatus = 1
+			GROUP By item_code, warehouse, parent
+			HAVING records > 1
+		""", new_name, as_dict=1)
+
+		if not records: return
+		document = "stock reconciliation" if len(records) == 1 else "stock reconciliations"
+
+		msg = _("The item {0} and {1} are present in the following {2} : <br>"
+			.format(frappe.bold(old_name), frappe.bold(new_name), document))
+
+		msg += ', '.join([get_link_to_form("Stock Reconciliation", d.parent) for d in records]) + "<br><br>"
+
+		msg += _("Note: To merge the items, create separate stock reconciliation for the old item {0}"
+			.format(frappe.bold(old_name)))
+
+		frappe.throw(_(msg), title=_("Can not merge {0} with {1}".format(old_name, new_name)))
```

Comment:
```suggestion
		frappe.throw(_(msg), title=_("Merge not allowed".format(old_name, new_name)))
```
Since we are mentioning items in the message itself, lets keep the title to the point and short. Less effort to read